### ISSUE TITLE
Deploying to Heroku fails

### ISSUE THREAD

**Issue by nicholasjhenry** [montrealelixir/website#39]
With the current build packs:

```
1. https://github.com/heroku/exec-buildpack
2. https://github.com/HashNuke/heroku-buildpack-elixir.git#560e1bd7d22edf8e4d0c6706dfbcaa94fa60c397
3. https://github.com/gjaldon/heroku-buildpack-phoenix-static.git#45e77b3b64996422e846d375a6f57e79d212488f
```

I get the following errors:

```
Counting objects: 47, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (46/46), done.
Writing objects: 100% (47/47), 6.51 KiB | 1.08 MiB/s, done.
Total 47 (delta 29), reused 0 (delta 0)
remote: Compressing source files... done.
remote: Building source:
remote:
remote: -----> Heroku Exec app detected
remote: -----> Elixir app detected
remote: -----> Checking Erlang and Elixir versions
remote:        Will use the following versions:
remote:        * Stack cedar-14
remote:        * Erlang 20.2
remote:        * Elixir 1.6.2
remote: -----> Will export the following config vars:
remote: DATABASE_URL
remote: MIX_ENV
remote: PAPERTRAIL_API_TOKEN
remote: POOL_SIZE
remote: SECRET_KEY_BASE
remote: SETUP_BY
remote: TWITTER_ACCESS_TOKEN
remote: TWITTER_ACCESS_TOKEN_SECRET
remote: TWITTER_CONSUMER_KEY
remote: TWITTER_CONSUMER_SECRET
remote: YOUTUBE_API_KEY
remote:        * MIX_ENV=prod
remote: -----> Fetching Erlang 20.2
remote: -----> Installing Erlang 20.2 (changed)
remote:
remote: -----> Fetching Elixir v1.6.2
remote: -----> Installing Elixir v1.6.2 (changed)
remote: [/app/tmp/cache/elixir-v1.6.2.zip]
remote:   End-of-central-directory signature not found.  Either this file is not
remote:   a zipfile, or it constitutes one disk of a multi-part archive.  In the
remote:   latter case the central directory and zipfile comment will be found on
remote:   the last disk(s) of this archive.
remote: unzip:  cannot find zipfile directory in one of /app/tmp/cache/elixir-v1.6.2.zip or
remote:         /app/tmp/cache/elixir-v1.6.2.zip.zip, and cannot find /app/tmp/cache/elixir-v1.6.2.zip.ZIP, period.
remote: chmod: cannot access ‘/tmp/build_0dce0583e9fdf341cb27d7ed890e0562/.platform_tools/elixir/bin/*’: No such file or directory
remote: -----> Installing Hex
remote: /app/tmp/buildpacks/1b25089e67cca4b4a2e95b2138863fedbf41f1a12ffbfe6f8684f630943dbaea6e1bce7e306959843e1c1ff5205f64c96ce12dc2bea0c25f57d646e4137ba0cd/lib/elixir_funcs.sh: line 87: mix: command not found
remote: -----> Installing rebar
remote: /app/tmp/buildpacks/1b25089e67cca4b4a2e95b2138863fedbf41f1a12ffbfe6f8684f630943dbaea6e1bce7e306959843e1c1ff5205f64c96ce12dc2bea0c25f57d646e4137ba0cd/lib/elixir_funcs.sh: line 93: mix: command not found
remote: -----> Fetching app dependencies with mix
remote: /app/tmp/buildpacks/1b25089e67cca4b4a2e95b2138863fedbf41f1a12ffbfe6f8684f630943dbaea6e1bce7e306959843e1c1ff5205f64c96ce12dc2bea0c25f57d646e4137ba0cd/lib/app_funcs.sh: line 47: mix: command not found
remote:  !     Push rejected, failed to compile Elixir app.
remote:
remote:  !     Push failed
remote: Verifying deploy...
remote:
remote: !       Push rejected to montreal-elixir-production.
remote:
To https://git.heroku.com/montreal-elixir-production.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'https://git.heroku.com/montreal-elixir-production.git'

```

Upgrade the buildpacks to:

```
https://github.com/HashNuke/heroku-buildpack-elixir.git#309c95e0bc0896ecf858696e813bced568c5eb09
https://github.com/gjaldon/heroku-buildpack-phoenix-static.git#45e77b3b64996422e846d375a6f57e79d212488f
```

We get the following errors:

```
remote: Compressing source files... done.
remote: Building source:
remote:
remote: -----> Phoenix app detected
remote:
remote: -----> Loading configuration and environment
remote:        Loading config...
remote:        Detecting assets directory
remote:        WARNING: no package.json detected in root nor custom directory
remote:        * assuming phoenix 1.3.x and later, please check config file
remote:        Will use phoenix configuration:
remote:        * assets path assets
remote:        * mix tasks namespace phx
remote:        Will use the following versions:
remote:        * Node 7.4.0
remote:        Will export the following config vars:
remote: DATABASE_URL
remote: MIX_ENV
remote: PAPERTRAIL_API_TOKEN
remote: POOL_SIZE
remote: SECRET_KEY_BASE
remote: SETUP_BY
remote: TWITTER_ACCESS_TOKEN
remote: TWITTER_ACCESS_TOKEN_SECRET
remote: TWITTER_CONSUMER_KEY
remote: TWITTER_CONSUMER_SECRET
remote: YOUTUBE_API_KEY
remote:        * MIX_ENV=prod
remote:
remote: -----> Installing binaries
remote:        Using cached node 7.4.0...
remote:        Installing Node 7.4.0...
remote:        Using default npm version
remote:
remote: -----> Building dependencies
remote:        Installing and caching node modules
remote: npm WARN package.json @ No description
remote: npm WARN package.json @ No README data
remote:        npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules/chokidar/node_modules/fsevents):
remote:        npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
remote:        npm WARN assets No description
remote:        npm WARN prefer global coffee-script@1.11.1 should be installed with -g
remote:        phoenix@1.3.0-rc.1 /tmp/build_6d110b9b098cf2238c9f1066c3a71be8/apps/montreal_elixir_web/assets/node_modules/phoenix
remote:        phoenix_html@2.9.3 /tmp/build_6d110b9b098cf2238c9f1066c3a71be8/apps/montreal_elixir_web/assets/node_modules/phoenix_html
...
remote:        npm WARN package.json @ No description
remote:        npm WARN package.json @ No README data
remote:        Running default compile
remote:        19:46:38 - info: compiled 8 files into 2 files, copied 7 in 2.5 sec
remote:        /app/tmp/buildpacks/023700e2ce6f1426de8dda8661743381070a2d8cfc005ee7b00ab2f7216a702dfccb64d4e962bf91fd83af603a8176d0c6aa885fb60a7976ef12f517409cb2cf/compile: line 5: mix: command not found
remote:  !     Push rejected, failed to compile Phoenix app.
remote:
remote:  !     Push failed
remote: Verifying deploy...
remote:
remote: !       Push rejected to montreal-elixir-production.
remote:
To https://git.heroku.com/montreal-elixir-production.git
 ! [remote rejected] master -> master (pre-receive hook declined)
error: failed to push some refs to 'https://git.heroku.com/montreal-elixir-production.git'
```

**Comment by saulecabrera** [comment]
Hey @nicholasjhenry,

Was taking a look at this, is there an specific reason to have the last two buildpacks pointing to those specific commits? 
I was able to reproduce the error when adding the buildpacks as they are in your example; when removing the hashes the deployment goes through.

See here: https://mtlelixir.herokuapp.com/

**Comment by nicholasjhenry** [comment]
Thank you for taking a look at this, @saulecabrera. We like to lock to a specific commit for the buildpacks so we know we are not floating on edge.  Would you be able to submit a PR with the SHA's that work? Admittedly, I did try this but didn't have any luck -- although I did it in a bit of rush :-)

**Comment by saulecabrera** [comment]
Sounds right @nicholasjhenry.

I've submitted #45; the only update needed is the elixir's buildpack sha. I've tested this on my own staging environment and it's working, let me know if it works on your end.

**Comment by nicholasjhenry** [comment]
Resolved in #45
